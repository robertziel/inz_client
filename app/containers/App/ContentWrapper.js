import styled from 'styled-components';

const ContentWrapper = styled.div`
  margin-left: 60px;
  transition: margin 100ms ease-out;

  &.sidebar-collapsed {
    margin-left: 260px;
    transition: margin 100ms ease-in;
  }
`;

export default ContentWrapper;
