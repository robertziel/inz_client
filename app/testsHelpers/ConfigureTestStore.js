import { memoryHistory } from 'react-router-dom';
import configureStore from 'configureStore';

export default function ConfigureTestStore() {
  const confuguredStore = configureStore({}, memoryHistory);
  this.store = confuguredStore.store;
  this.presistor = confuguredStore.presistor;
}
