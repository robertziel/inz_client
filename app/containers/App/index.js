/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { connect } from 'react-redux';
import { Switch, Route } from 'react-router-dom';

import PropTypes from 'prop-types';
import { createSelector } from 'reselect';

import Navbar from 'components/Navbar/index';
import Footer from 'components/Footer/index';
import Sidebar from 'components/Sidebar/index';
import { sidebarToggleSelector } from 'components/Sidebar/selectors';

import HomePage from 'containers/_pages/HomePage/Loadable';
import ProfilePage from 'containers/_pages/ProfilePage/Loadable';
import NotFoundPage from 'containers/_pages/NotFoundPage/Loadable';

import ContentWrapper from './ContentWrapper';

function App(props) {
  return (
    <div>
      <Navbar />
      <ContentWrapper
        className={props.sidebarCollapsed ? 'sidebar-collapsed' : ''}
      >
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/profile" component={ProfilePage} />
          <Route component={NotFoundPage} />
        </Switch>
        <Footer />
      </ContentWrapper>
      <Sidebar />
    </div>
  );
}

App.propTypes = {
  sidebarCollapsed: PropTypes.bool.isRequired,
};

function mapStateToProps() {
  return createSelector(
    sidebarToggleSelector(),
    sidebarCollapsed => ({
      sidebarCollapsed,
    }),
  );
}

export default connect(mapStateToProps)(App);
