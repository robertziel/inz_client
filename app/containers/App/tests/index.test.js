/* global context */

import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ConfigureTestStore from 'testsHelpers/ConfigureTestStore';

import App from '../index';

function shallowWrapper() {
  return shallow(<App store={store} />)
    .find('App')
    .shallow();
}

let store;
let wrapper;

beforeEach(() => {
  store = new ConfigureTestStore().store;
  wrapper = shallowWrapper();
});

describe('<App />', () => {
  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  context('when sidebar collapses', () => {
    beforeEach(() => {
      // It is better to update redux state than use setProps but it does
      // not work according to https://github.com/airbnb/enzyme/issues/2009
      wrapper.setProps({ sidebarCollapsed: true });
    });

    it('should add class to <ContentWrapper />', () => {
      expect(wrapper.exists('.sidebar-collapsed')).toBe(true);
    });
  });
});
