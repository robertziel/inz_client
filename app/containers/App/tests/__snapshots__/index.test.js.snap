// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div>
  <Navbar />
  <ContentWrapper
    className=""
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/profile"
      />
      <Route
        component={[Function]}
      />
    </Switch>
    <Footer />
  </ContentWrapper>
  <Connect(Sidebar) />
</div>
`;
